### ISSUE TITLE
FR-T5.2-13: Template library publishing service should be RADON IAM compliant

### ISSUE THREAD

**Issue by neza-vehovar** [radon-h2020/radon-delivery-toolchain#42]
ID: | FR-T5.2-13
-- | --
Type: | USABILITY
User   story: | As a user I would like to use the same credentials for the whole RADON project including template library publishing service.
Requirement: | The TLPS needs to support the use of agreed upon security measures.  
Extended   Description: | The TLPS needs to support the use of general RADON credentials as CLI login credentials.
Priority: | Should have
Affected   Tools: | TEMPLATE_LIBRARY
Means   of Verification: | Using template library CLI and template library API with RADON IAM credentials.

**Comment by cankarm** [comment]
This is a new requirement raised by user request, who wants to have one ID management inside RADON. 

@gcasale , please confirm and remove "WorkInProgress" label, if OK.

**Comment by gcasale** [comment]
Approved.

**Comment by cankarm** [comment]
Publishing to TPS is possible with RADON credentials (RADON IDE KeyCloak support) at least for Eclipse Che plugin.

**Comment by anzoman** [comment]
Authentication to TPS can now be performed using RADON KeyCloak/Che. This can be done when using TPS REST API where you will be redirected to login as a RADON KeyCloak/Che user or even better you can use [Eclipse Che TPS plugin](https://github.com/radon-h2020/radon-template-library-publishing-service-plugin) to be able tu use TPS actions within RADON IDE/Eclipse Che. More info about TPS auth is available here: https://template-library-radon.xlab.si/docs/documentation.html#keycloak-user-login-and-authentication.

@cankarm I believe that this is done and can be closed.

**Comment by cankarm** [comment]
@anzoman  what about TPS CLI? Is it possible to use RADON/Che KeyCloak user?

**Comment by anzoman** [comment]
@cankarm no, on TPS CLI we still need to implement this. Currently the login is possible only through our KeyCloak.

**Comment by anzoman** [comment]
@cankarm the feature to login with RADON IDE KeyCloak credentials via [Template library CLI](https://pypi.org/project/xopera-template-library/) is now available in version [`0.2.1`](https://pypi.org/project/xopera-template-library/0.2.1/).

The login process with my RADON KeyCloak credentials is shown below:

```console
(.venv) anzoman@ubuntu:~/Desktop/template-library-cli$ xopera-template-library setup --keycloak
Current Template Library REST API endpoint: https://template-library-radon.xlab.si/api
Modify API endpoint (press enter to keep the current): 
Template library REST API endpoint has been set to: https://template-library-radon.xlab.si/api.

Current KeyCloak auth endpoint: https://openid-radon.xlab.si
Modify KeyCloak endpoint (press enter to keep the current): 
KeyCloak endpoint has been set to: https://openid-radon.xlab.si.

(.venv) anzoman@ubuntu:~/Desktop/template-library-cli$ xopera-template-library login --keycloak radon
Username: anze.luzar@xlab.si                                                    
Password: 
User 'anze.luzar@xlab.si' logged in.

(.venv) anzoman@ubuntu:~/Desktop/template-library-cli$ xopera-template-library user info 
+User data-----------+--------------------+--------------------+---------------------+
| ID | Name          | Username           | Email              | Created at          |
+----+---------------+--------------------+--------------------+---------------------+
| 6  | KeyCloak user | anze.luzar@xlab.si | anze.luzar@xlab.si | 2020-10-29 14:55:39 |
+----+---------------+--------------------+--------------------+---------------------+
```

**Comment by cankarm** [comment]
Perfect! Thanks @anzoman !
